enacts/wat_bal/maproom_monit.py

```diff
@@ -0,0 +1,512 @@
+    Input("planting_month", "value"),
+    Input("wat_bal_plot", "clickData"),
+    State("time_selection", "options"),
+)
+def update_time_sel(planting_day, planting_month, graph_click, current_options):
+    if dash.ctx.triggered_id == "wat_bal_plot":
+        time_range = current_options
+        the_value = graph_click["points"][0]["x"]
+    else:
+        time_range = rr_mrg.precip["T"].isel({"T": slice(-366, None)})
+        p_d = time_range.where(
+            lambda x: (x.dt.day == int(planting_day))
+                & (x.dt.month == calc.strftimeb2int(planting_month)),
+            drop=True
+        ).squeeze()
+        time_range = time_range.where(
+            time_range >= p_d, drop=True
+        ).dt.strftime("%-d %b %y").values
+        the_value = time_range[-1]
+    return time_range, the_value
+
+
+@APP.callback(
+    Output("layers_control", "children"),
+    Input("map_choice", "value"),
+    Input("time_selection", "value"),
+    Input("submit_kc", "n_clicks"),
+    State("planting_day", "value"),
+    State("planting_month", "value"),
+    State("kc_init", "value"),
+    State("kc_init_length", "value"),
+    State("kc_veg", "value"),
+    State("kc_veg_length", "value"),
+    State("kc_mid", "value"),
+    State("kc_mid_length", "value"),
+    State("kc_late", "value"),
+    State("kc_late_length", "value"),
+    State("kc_end", "value"),
+)
+def make_map(
+        map_choice,
+        the_date,
+        n_clicks,
+        planting_day,
+        planting_month,
+        kc_init,
+        kc_init_length,
+        kc_veg,
+        kc_veg_length,
+        kc_mid,
+        kc_mid_length,
+        kc_late,
+        kc_late_length,
+        kc_end,
+):
+    qstr = urllib.parse.urlencode({
+        "map_choice": map_choice,
+        "the_date": the_date,
+        "n_clicks": n_clicks,
+        "planting_day": planting_day,
+        "planting_month": planting_month,
+        "kc_init": kc_init,
+        "kc_init_length": kc_init_length,
+        "kc_veg": kc_veg,
+        "kc_veg_length": kc_veg_length,
+        "kc_mid": kc_mid,
+        "kc_mid_length": kc_mid_length,
+        "kc_late": kc_late,
+        "kc_late_length": kc_late_length,
+        "kc_end": kc_end,
+    })
+    return [
+        dlf.BaseLayer(
+            dlf.TileLayer(
+                url="https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png",
+            ),
+            name="Street",
+            checked=False,
+        ),
+        dlf.BaseLayer(
+            dlf.TileLayer(
+                url="https://{s}.tile.opentopomap.org/{z}/{x}/{y}.png"
+            ),
+            name="Topo",
+            checked=True,
+        ),
+    ] + [
+        make_adm_overlay(
+            adm["name"],
+            adm["sql"],
+            adm["color"],
+            i+1,
+            len(CONFIG_GLOBAL["shapes_adm"])-i,
+            is_checked=adm["is_checked"]
+        )
+        for i, adm in enumerate(CONFIG_GLOBAL["shapes_adm"])
+    ] + [
```

Comment:
This is outside the scope of this PR, but I like the admin level chooser in the FBF maproom better than this mechanism. There are too many possible configurations here, most of which don't make sense. Typically each region at level n is the union of the smaller regions at level n+1, so if level n+1 is enabled then the boundary of level n is visible regardless of whether level n is enabled or not. Do you have a use in mind where selecting a single admin level would not support what you want to do?

Reply:
There are cases where the n+1 level is not a partition of level n, but that's due to lack of comprehensive and coherent sources rather than on purpose. So we can leave that aside.

Then yes, if one is trying to localize an entity at level n, it helps a great deal to map an overlay of levels m smaller than n, typically thicker, possibly in stronger color. We do that all over.

Reply:
> if one is trying to localize an entity at level n, it helps a great deal to map an overlay of levels m smaller than n, typically thicker, possibly in stronger color.

Indeed, the mode control in the FbF application doesn't support that. I'm not sure how this setup supports it either, though. If there are four admin levels available and the user chooses to display all of them, how do you know which level they're trying to select? Assume the highest of the displayed levels? I guess that's a reasonable assumption, but it seems a bit complicated for the user. E.g. if we start with all levels displayed, and I want to display something from level 2, I need to uncheck level 1. That doesn't seem obvious to me.

How does it work in classic maprooms?

The best solution I can think of at the moment is to display all levels always, and have a control for selection, not for display. Or make it zoom-dependent, so we display all levels that can be meaningfully displayed at the current resolution.

Reply:
ah... I see what the deal is now. I am thinking only display here, not about selection. None of the Maprooms I worked on select other than grid boxes. In classic Maprooms, the 2 are independent: you display whatever, you use a menu to tell which level you want to select against (even if not displayed), you click the map or pick a name in a menu to select.

I believe that dash-leaflet allows only to select geometries that are displayed, correct? In that case, what we might need is:
1/ a menu where the user tells at what level they want to select (gridbox, admin3, admin2, admin1)
2/ have that selection trigger displaying the selected level
3/ click the map, get multiple inputs (whichever layers are displayed) and use the info in the menu in 1/ to get the one we want

Maybe?

Reply:
You're saying two separate controls, one for display (pick as many as you want) and one for selection (pick one)? That at least seems comprehensible. And as long as the default is to display all levels, it doesn't seem intrusive. (We may have to be smarter about choosing line thicknesses relative to zoom level for displaying all levels to be a good default.)

Reply:
That's what Classic does. Checkboxes for the different layers of the map, in a specific Layers control. Another drop down menu to pick what gets selected when clicking map.
I would rather say as long as the default displays at least the level that is the default to pick. Displaying all levels can be overwhelming the map, I'd say from 4 it's no good.
I noticed that thickness was responding to zoom level. And I also noticed the order of the admin layers depends of the order they were selected (or something more confusing). It's usually better to the eye that the lower admin levels are on top and hiding their higher level counterpart.

Reply:
OK. Make this change here or make an issue for later, as you wish. Resolve this conversation when you're done with it.

Reply:
Oh, there's no selection here, so there's nothing to change?

Reply:
haha ! yes, selections are done on grid boxes.